### ISSUE TITLE
Profile editor does not respect profile properties order

### ISSUE THREAD

**Issue by valadas** [dnnsoftware/Dnn.Platform#2542]
## Description of bug
When changing the profile properties order, this change is not reflected when users edit their profile.

## Steps to reproduce
List the steps to reproduce the behavior:
1. Go to Site Settings -> Site Behaviour -> User Profiles
2. Change some profile properties order ( I reversed the order of Country and Region)
3. Close the Persona Bar
4. Click on the user name on the skin, then edit profile

## Current result
The change of the ordering is not reflected

## Expected result
The profile editor should follow the profile properties order

## Additional context
The sort order does change in the database table ProfilePropertyDefinition
On a new installation of 9.3.0 (RC0), the order is Country before Region as defined int the blank and default site templates as per https://github.com/dnnsoftware/Dnn.Platform/pull/2260 but then there is no way to re-change the order in the UI (it changes but is not reflected)

## Affected version

* [x] 9.3.0
* [x] 9.2.2
* [x] 9.2.1
* [x] 9.2
* [x] 9.1.1
* [ ] 9.1
* [ ] 9.0

**Comment by valadas** [comment]
Yes, the goal of #2146 and #2260 was for having the Country before the Region as you explain. It was implemented in a way that would only do it on new sites in order to not affect existing sites. While testing 9.3.0 RC0 I noticed that the order set is not respected everywhere. But there where issues with the Sortable component too, not sure if still an issues now, waiting on RC1, will retest and update this issue accordingly.

**Comment by valadas** [comment]
It is still the case in 9.4.0 RC

**Comment by Timo-Breumelhof** [comment]
@valadas I have added and edited some Profile Property Definitions in SiteSettings > User Profiles and the Editor has some issues I noticed (in an ML environment).
Drag and drop order issues and empty fields when when switching to the Localized values screen.
This issue is not about that, right?
